sycl/include/CL/sycl/detail/builtins.hpp

```diff
@@ -16,219 +16,233 @@
 // TODO Decide whether to mark functions with this attribute.
 #define __NOEXC /*noexcept*/
 
-#define MAKE_CALL_ARG1(call)                                                   \
+#ifdef __SYCL_DEVICE_ONLY__
+#define __FUNC_PREFIX_OCL  __spirv_ocl_
+#define __FUNC_PREFIX_CORE  __spirv_
+#else
+#define __FUNC_PREFIX_OCL
+#define __FUNC_PREFIX_CORE
+#endif
+
+#define PPCAT_NX(A, B) A ## B
+#define PPCAT(A, B) PPCAT_NX(A, B)
```

Comment:
Perhaps adding an `#undef` at the end of the file for both of them too?

Reply:
I've added `#undef`-s. Thanks!